internal/git/gitsync_processor.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2023 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"errors"
+	"log"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/numaproj-labs/numaplane/api/v1"
+)
+
+type Message struct {
+	Updates bool
+	Err     error
+}
+
+type GitSyncProcessor struct {
+	gitSync  v1.GitSync
+	k8Client client.Client
+	channels []chan Message
+}
+
+func checkForUpdates(r *git.Repository) (bool, error) {
+	err := r.Fetch(&git.FetchOptions{
+		RemoteName: "origin",
+	})
+	if err != nil && !errors.Is(err, git.NoErrAlreadyUpToDate) {
+		return false, err
+	}
+	// Get local and remote references
+	localRef, err := r.Head()
+	if err != nil {
+		return false, err
+	}
+	remoteRef, err := r.Reference(plumbing.NewRemoteReferenceName("origin", "master"), true)
+	if err != nil {
+		return false, err
+	}
+	// Compare the references
+	if localRef.Hash() != remoteRef.Hash() {
+		return true, nil
+	}
+	return false, nil
+}
+
+func WatchRepo(repoPath v1.RepositoryPath, ch chan Message) {
+	// clone the repository
+	repo, err := git.PlainClone(repoPath.Path, false, &git.CloneOptions{
```

Comment:
Why do you pass in `repoPath.Path` here? shouldn't it be a local path? Moreover, I'm not sure if we want to persist the change. Maybe we can use the in memory cloning ATM. @juliev0 any thoughts?

Reply:
Just looking at the go-git library myself for the first time. I guess you're more the expert on this @xdevxy so feel free to advise. I guess it does take a local path - but perhaps that could be equal to the `repo.Path`? 

I didn't realize there was an in-memory cloning option until now - are you thinking that will be faster then? That seems good if so.